### PR TITLE
enhancement(loki sink): Internal Events updates

### LINKED ISSUE #10401: (loki sink): Update internal events
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Current Vector Version

```
❯ vector --version
vector 0.18.1 (x86_64-apple-darwin c4adb60 2021-11-30)
```

### Proposal

Loki's OutOfOrder internal events are both logged at WARN and emit `processing_error` counters, I'd propose when we emit these logs they are the _intended_ configured behavior.

When rewriting the timestamp it's not necessarily a processing error as the event is processed and sent to Loki. Dropping an event emits both a discarded event and processing error metric.

I'd propose neither log should be WARN, and the metrics could be updated to be more accurate/user friendly.

### References

- #10364

cc @jszwedko @bruceg @binarylogic

### PR DESCRIPTION
**Pull request by spencergilbert**
- enhancement(loki sink): emit OutOfOrder logs at debug level
- remove processing error metrics, add rewritten timestamp counter
- add deprecation comments and new component_discarded_events_total metric

Closes #10401

~This needs doc updates + release notes, will add when changes are~ 👍

### CODE REVIEW

File: src/internal_events/loki.rs [hunk 1]
```diff
@@ -36,17 +36,16 @@ pub struct LokiOutOfOrderEventDropped;
...
         );
     }
 
     fn emit_metrics(&self) {
         counter!("events_discarded_total", 1,
-                "reason" => "out_of_order");
-        counter!("processing_errors_total", 1,
-                "error_type" => "out_of_order");
+                "reason" => "out_of_order"); // deprecated
+        counter!("component_discarded_events_total", 1);
```
Comment (spencergilbert) [inline src/internal_events/loki.rs:48]: I didn't include the `reason` here as that's the only reason that we drop events in this sink (AFAIK)

File: src/internal_events/loki.rs [hunk 2]
```diff
@@ -55,14 +54,13 @@ pub struct LokiOutOfOrderEventRewritten;
...
+        debug!(
             message = "Received out-of-order event, rewriting timestamp.",
             internal_log_rate_secs = 30
         );
     }
 
     fn emit_metrics(&self) {
-        counter!("processing_errors_total", 1,
-                "error_type" => "out_of_order");
+        counter!("rewritten_timestamps_total", 1);
```
Comment (spencergilbert) [inline src/internal_events/loki.rs:64]: Not sold on the metric name...

File: src/internal_events/loki.rs (hunk 1 above)
Comment (bruceg) [reply src/internal_events/loki.rs:48]: I think it's still a good idea to include it to make it obvious this is not due to some other error.

File: src/internal_events/loki.rs (hunk 2 above)
Comment (bruceg) [reply src/internal_events/loki.rs:64]: Yeah, we don't have a precedent or standard for this. I don't have a strong opinion, other than it should probably have a suffix of `_events_total` (include units).

File: src/internal_events/loki.rs [hunk 3]
```diff
@@ -36,17 +36,17 @@ pub struct LokiOutOfOrderEventDropped;
...
         );
     }
 
     fn emit_metrics(&self) {
         counter!("events_discarded_total", 1,
+                "reason" => "out_of_order"); // deprecated
+        counter!("component_discarded_events_total", 1,
                 "reason" => "out_of_order");
-        counter!("processing_errors_total", 1,
-                "error_type" => "out_of_order");
```
Comment (jszwedko) [inline src/internal_events/loki.rs:49]: We should call out in the upgrade guide that this deprecated metric was dropped.

**Review by jszwedko — COMMENTED** [review]
Looks good!

File: src/internal_events/loki.rs (hunk 3 above)
Comment (spencergilbert) [reply src/internal_events/loki.rs:49]: 🤔 Do we want to just leave this deprecated this release and drop it next release?

Comment (jszwedko) [reply src/internal_events/loki.rs:49]: Yeah, that would be preferable: to go through the deprecation cycle.

Comment (spencergilbert) [reply src/internal_events/loki.rs:49]: IDK why I didn't do that to begin with...

File: website/content/en/highlights/2022-02-08-0-20-0-upgrade-guide.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 date: "2022-02-08"
 title: "0.20 Upgrade Guide"
 description: "An upgrade guide that addresses breaking changes in 0.20.0"
-authors: ["topbz"]
+authors: ["topbz","spencergilbert"]
```
Comment (bruceg) [inline website/content/en/highlights/2022-02-08-0-20-0-upgrade-guide.md:5]: Is `topbz` right and not `tobz`??

**Review by bruceg — APPROVED** [review]
LGTM except for a question about the author handle.

File: website/content/en/highlights/2022-02-08-0-20-0-upgrade-guide.md (hunk 4 above)
Comment (spencergilbert) [reply website/content/en/highlights/2022-02-08-0-20-0-upgrade-guide.md:5]: Pfff, if @tobz wants to misspell his handle that's his choice...

Comment (tobz) [reply website/content/en/highlights/2022-02-08-0-20-0-upgrade-guide.md:5]: 🤣 

I wouldn't _mind_ if you wanted to fix it for me. :)

**Review by jszwedko — COMMENTED** [review]
Just a reminder to re-add the removed metrics and mark them as deprecated in the upgrade guide.

**Comment by spencergilbert** [comment]
> Just a reminder to re-add the removed metrics and mark them as deprecated in the upgrade guide.

Doing that today 👍

**Pull request merged**